Pipeline for annotating genomes using long read transcriptomics data with stringtie and other tools
===================================================================================================

This `snakemake` pipeline creates GFF annotation from Oxford Nanopore cDNA or direct RNA reads.
The cDNA reads are optionally processed by [pychopper](https://github.com/nanoporetech/pychopper) for trimming and orientation. The processed reads are mapped to the reference genome using [minimap2](https://github.com/lh3/minimap2), and then processed by [stringtie](http://ccb.jhu.edu/software/stringtie) in long read mode (with or withouth using a guide annotation) to generate the GFF annotation.

The annotation generated by the pipeline is compared to the existing annotation (if one can be found) using [gffcompare](http://ccb.jhu.edu/software/stringtie/gffcompare.shtml) and a report is genereated.

Getting Started
===============

## Input

- The input reads must be in fastq format. 
- The input genome must be in fasta format.
- The existing annotations must be in GFF3 or GFF2 format.

## Output

The main output files generated by the pipeline are the following:

- `results/annotation/str_merged.gff` - annotation in [GFF2](http://gmod.org/wiki/GFF2) format
- `results/gffcompare/*` - comparison to the existing annotation (if provided) by the gffcompare tool ([documentation](https://ccb.jhu.edu/software/stringtie/gffcompare.shtml))
- `results/gffcompare/str_gffcmp_report.pdf` - visualisation of `gffcompare` results

## Depedencies

- [miniconda](https://conda.io/miniconda.html)
- [snakemake](http://snakemake.readthedocs.io/en/latest/) - easily installed via conda
- The rest of the dependencies are installed via conda.

## Installation

Clone the pipeline and the pipeline toolset by issuing:

```bash
git clone --recursive https://github.com/nanoporetech/pipeline-ref-isoforms-ONT.git
```

## Usage

Edit `config.yml` to set the input genome, input fastq and parameters, then issue:

```bash
snakemake --use-conda -j <num_cores> all
```